### PR TITLE
JAMES-3693 Implement RedisExtension for testing

### PR DESCRIPTION
**Pull request by vttranlina**
https://issues.apache.org/jira/browse/JAMES-3693
RedisExtension will help us when writing unit tests, integration-test the email-limiting feature with Redis implement.

### CODE REVIEW

File: server/mailet/rate-limiting-redis/pom.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    <parent>
+        <groupId>org.apache.james</groupId>
+        <artifactId>james-server</artifactId>
+        <version>3.7.0-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>james-server-rate-limiting-redis</artifactId>
+    <packaging>jar</packaging>
+    <name>Apache James :: Server :: Mailet :: Rate limiting</name>
```
Comment (chibenwa) [inline server/mailet/rate-limiting-redis/pom.xml:31]:
```suggestion
    <name>Apache James :: Server :: Mailet :: Rate limiting Redis</name>
```

File: server/mailet/rate-limiting-redis/src/test/java/org/apache/james/ratelimiting/DockerRedisRule.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.ratelimiting;
+
+import org.apache.james.GuiceModuleTestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+public class DockerRedisRule implements GuiceModuleTestRule {
```
Comment (chibenwa) [inline server/mailet/rate-limiting-redis/src/test/java/org/apache/james/ratelimiting/DockerRedisRule.java:26]: Please drop `DockerRedisRule` that is for Junit 4 and concentrate solely on the extension that is Junit 5

File: server/mailet/rate-limiting-redis/src/test/java/org/apache/james/ratelimiting/RedisExtension.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+package org.apache.james.ratelimiting;
+
+import org.apache.james.GuiceModuleTestExtension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+
+public class RedisExtension implements GuiceModuleTestExtension {
+    private final DockerRedisRule dockerRedisRule = new DockerRedisRule();
```
Comment (chibenwa) [inline server/mailet/rate-limiting-redis/src/test/java/org/apache/james/ratelimiting/RedisExtension.java:28]: Can we use a singletong to run the docker redis container?

This allow to reuse the docker container accross test classes running on the same surefire JVM processes.

File: server/mailet/rate-limiter-redis/pom.xml [hunk 4]
```diff
@@ -26,9 +26,9 @@
         <relativePath>../../pom.xml</relativePath>
     </parent>
 
-    <artifactId>james-server-rate-limiting-redis</artifactId>
+    <artifactId>james-server-rate-limiter-redis</artifactId>
     <packaging>jar</packaging>
-    <name>Apache James :: Server :: Mailet :: Rate limiting</name>
+    <name>Apache James :: Server :: Mailet :: Rate limiting Redis</name>
```
Comment (chibenwa) [inline server/mailet/rate-limiter-redis/pom.xml:31]:
```suggestion
    <name>Apache James :: Server :: Mailet :: Rate limiting :: Redis</name>
```

File: server/mailet/rate-limiter-redis/src/test/java/org/apache/james/rate/limiter/RedisExtension.java [hunk 5]
```diff
@@ -17,28 +17,28 @@
...
+package org.apache.james.rate.limiter;
 
 import org.apache.james.GuiceModuleTestExtension;
 import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.jupiter.api.extension.ParameterContext;
 import org.junit.jupiter.api.extension.ParameterResolutionException;
 
 public class RedisExtension implements GuiceModuleTestExtension {
-    private final DockerRedisRule dockerRedisRule = new DockerRedisRule();
+    public static final DockerRedis DOCKER_REDIS_SINGLETON = new DockerRedis();
```
Comment (chibenwa) [inline server/mailet/rate-limiter-redis/src/test/java/org/apache/james/rate/limiter/RedisExtension.java:28]: Why public?

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**